### PR TITLE
Adds docs about live migration

### PR DESCRIPTION
**Pull request by stevsmit**
Version(s):

Issue:

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
+= Live migration to the OVN-Kubernetes network plugin overview
+
+A _live_ migration process is available when migrating to the OVN-Kubernetes network plugin from the OpenShift SDN plugin. This migration method offers no service interruption, minimizes downtown, and automatically reboots nodes. This method is valuable for deployments types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS.
```
Comment (martinkennelly) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:3]: I think define what live means here

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+* With the live migration, there is a Maximum Transmission Unit (MTU) mismatch between OpenShift SDN and OVN-Kubernetes, and MTU values might differ. The Cluster Network Operator (CNO) handles updating the routable MTU to ensure both CNIs share the same overlay MTU. 
+
+* When the live migration procedure begins, multitenant isolation is permanently disabled. Multitenant isolation is only supported on OpenShift SDN. 
+
+* During the live migration, when the cluster is running both OVN-Kubernetes and OpenShift SDN, multicast and egress IP are temporarily disabled for both CNIs. The egress firewall remains functional during this process.
+
+* Although a rollback procedure is provided, the migration is intended to be a one-way process. For users that want to rollback to OpenShift-SDN, migration from OpenShift-SDN to OVN-Kubernetes must have succeeded. Users can follow the same procedure below to migrate to the OpenShift SDN network plugin from the OVN-Kubernetes network plugin.
+
+* Live migration does not work in clusters without the `dedicated-admin` namespace. This namespace is only created by _managed_ clusters.
```
Comment (martinkennelly) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:15]: id remove this implementation detail about namespaces being present. 
Just that live migration only support on  managed clusters excluding hypershift based clusters.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+* You have created a recent backup of the etcd database is available.
+
+* The cluster is in a known good state without any errors.
+
+* On all cloud platforms after updating software, a security group rule must be in place to allow UDP packets on port `6081` for all nodes.
+
+* Cluster administrators must the following custom resources:
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:38]: cc @weliang1 I'm not sure of the answer to this question.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
+= Live migration to the OVN-Kubernetes network plugin overview
+
+A _live_ migration process is available when migrating to the OVN-Kubernetes network plugin from the OpenShift SDN plugin. This migration method offers no service interruption, minimizes downtown, and automatically reboots nodes. This method is valuable for deployments types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS.
+
+== Considerations for migrating to the OVN-Kubernetes network plugin
+
+* With the live migration, there is a Maximum Transmission Unit (MTU) mismatch between OpenShift SDN and OVN-Kubernetes, and MTU values might differ. The Cluster Network Operator (CNO) handles updating the routable MTU to ensure both CNIs share the same overlay MTU.
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:7]: Removed.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* 
+
+= Migrating to the OVN-Kubernetes network plugin (live migration)
+
+Use the following procedure to migrate from OpenShift SDN to OVN-Kubernetes
+
+////
+.Procedure
+
+. Edit the `Network.operator.openshift.io` custom resource (CR) by entering the following command:
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:95]: I think that the procedure is: 

## SDN to OVN migration
oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"unsupported-red-hat-internal-testing": "true"}}}' 
oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'

## OVN rollback to SDN
oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OpenShiftSDN"}}'

So these commands aren't necessary ?

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 6]
```diff
@@ -0,0 +1,166 @@
+= OpenShift SDN live migration overview
+
+The OpenShift SDN live migration is the process in which the OpenShift SDN network plugin and its network configurations, connections, and associated resources, are migrated to the OVN-Kubernetes network plugin. The live migration method offers the following benefits: 
+
+* Continuous service availability
+* Minimized downtime
+* Automatic node rebooting
+* Seamless transition from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin. 
+
+The live migration method is valuable for deployment types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS. It is not available for HyperShift deployment types.
```
Comment (weliang1) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:10]: @stevsmit The migration is not supported in ARO yet in 4.15 due to https://issues.redhat.com/browse/ARO-4352

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+The live migration method is valuable for deployment types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS. It is not available for HyperShift deployment types. 
+
+== Considerations for migrating to the OVN-Kubernetes network plugin
+
+* When the live migration procedure begins, multitenant isolation is permanently disabled. Multitenant isolation is only supported on OpenShift SDN. 
+
+* During the live migration, when the cluster is running both OVN-Kubernetes and OpenShift SDN, multicast and egress IP are temporarily disabled for both CNIs.
+
+* Although a rollback procedure is provided, the migration is intended to be a one-way process. For users that want to rollback to OpenShift-SDN, migration from OpenShift-SDN to OVN-Kubernetes must have succeeded. Users can follow the same procedure below to migrate to the OpenShift SDN network plugin from the OVN-Kubernetes network plugin.
```
Comment (weliang1) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:18]: I assume that the statement"the migration is intended to be a one-way process" is from offline migration docs prior to version 4.15, I am not sure this is still be true for live migration in 4.15. @pliurh can confirm it.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 6 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:10]: Fixed this one.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 7 above)
Comment (pliurh) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:18]: The statement is still valid for live migration. The purpose of the SDN migration (either offline or live) is to give customers a way to drop SDN.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+* The live migration is only supported on managed clusters. It is not supported on HyperShift clusters. 
+
+== Information about the OpenShift SDN live migration script
+
+The live migration method for migrating from OpenShift SDN network plugin to the OVN-Kubernetes network plugin is a two-step procedure that contains multiple internal processes that are executed by a wrapper script. When running the two commands, the following events occur:
```
Comment (pliurh) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:24]: The live migration is a multi-step procedure that resides in the CNO code.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+. The Cluster Network Operator checks the current hardware MTU and the cluster MTU to determine a routable MTU during migration. These changes trigger the Machine Config Operator to apply a new machine configuration to each machine config pool and reboot all nodes:
+
+** If the current hardware MTU is `1500`, and the cluster MTU is not specified, the CNO sets the `.spec.migration.mtu` of the `network.operator` CR with the MTU of OVN-Kubernetes: 
+*** `.spec.migration.mtu.network.from` is reduced from `1450` to `1400`
+*** `.spec.migration.mtu.machine.from` remains `1500`
+
+** If the current hardware MTU is `9000`, and the cluster MTU is specified explicitly to `1250` for OpenShift SDN, the CNO sets the `.spec.migration.mtu` of the `network.operator` CR with the MTU of OVN-Kubernetes:
+*** `.spec.migration.mtu.network.from` is reduced from `1250` to `1200`
+*** `.spec.migration.mtu.network.from` stays `9000`
```
Comment (pliurh) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:36]: These are just examples. Users shall be aware that the cluster MTU will be reduced by 50 bytes after the migration.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+* The cluster is in a known good state without any errors.
+
+* On all cloud platforms after updating software, a security group rule must be in place to allow UDP packets on port `6081` for all nodes.
+
+* Cluster administrators must disable the following custom resources:
++
+** Egress Router
+** Egress IP
+** Egress Firewalls
+** Multicast
```
Comment (pliurh) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:88]: Only the egress router configuration shall be removed. The configuration for other features can be migrated automatically after the live migration is done. The other features just don't work during the live migration.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+* link:https://access.redhat.com/solutions/4854561[Removing an egress IP address]
+* To remove an egress router, delete the pod. 
+
+= Using the live migration method to migrate from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin
+
+Administrators should ensure that all prerequisites have been met before beginning the migration process. Afterwards, the following procedure can be used to initiate the live migration from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin.
+
+.Procedure
+
+. Enter the following command to add the `unsupported-red-hat-internal-testing` annotation to the cluster-level network configuration:
```
Comment (pliurh) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:140]: @martinkennelly Was it added when I was on PTO?

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+----
+
+. Enter the following command to patch the cluster-level networking configuration and initiative the migration from OpenShift SDN to OVN-Kubernetes:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+----
+
+After running these commands, the migration process begins. The time it takes to complete the process will vary depending on how many nodes are in your cluster.
```
Comment (pliurh) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:154]: We shall ask users to monitor the `status.conditions` of the `network.config` CR. Additionally, the annotation `network.openshift.io/network-type-migration` can be removed when the migration is completed.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+The OpenShift SDN live migration is the process in which the OpenShift SDN network plugin and its network configurations, connections, and associated resources, are migrated to the OVN-Kubernetes network plugin. The live migration method offers the following benefits: 
+
+* Continuous service availability
+* Minimized downtime
+* Automatic node rebooting
+* Seamless transition from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin. 
+
+The live migration method is valuable for deployment types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS. It is not available for HyperShift deployment types. 
+
+== Considerations for migrating to the OVN-Kubernetes network plugin
```
Comment (kyrtapz) [inline modules/nw-ovn-kubernetes-migration-about-live.adoc:12]: We should add a node that no new nodes should be created during the live migration, this is not supported as both CNIs would try to setup the new node at once.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 11 above)
Comment (martinkennelly) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:140]: I didnt add it and I didnt see anyone else add it. I dont know where this annotation came from.

Comment (kyrtapz) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:140]: that's the annotation required by SD.
Currently there is a webhook which will not allow for live-migration to start unless this annotation is set, this was done to ensure users do not try out live migration before we conclude it all of our testing.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (weliang1) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc]: @martinkennelly @stevsmit Check the response in https://github.com/openshift/openshift-docs/pull/71390#discussion_r1497095891 from @pliurh

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+* When the live migration procedure begins, multitenant isolation is permanently disabled. Multitenant isolation is only supported on OpenShift SDN.
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:14]: Updated with: 

* The live migration procedure is unsupported for clusters with OpenShift SDN multitenant mode enabled. You must disable multitenant mode before beginning the process.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+* Automatic node rebooting
+* Seamless transition from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin. 
+
+The live migration method is valuable for deployment types that require constant service availability, such as Azure Red Hat OpenShift (ARO), Red Hat OpenShift Dedicated, or Red Hat OpenShift Service on AWS. It is not available for HyperShift deployment types. 
+
+== Considerations for migrating to the OVN-Kubernetes network plugin
+
+* When the live migration procedure begins, multitenant isolation is permanently disabled. Multitenant isolation is only supported on OpenShift SDN. 
+
+* During the live migration, when the cluster is running both OVN-Kubernetes and OpenShift SDN, multicast and egress IP are temporarily disabled for both CNIs.
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:16]: Updated with: 

* Multicast, egress IP addresses, egress routers, and egress firewalls will block the live migration process. These features must be disabled or removed prior to beginning the live migration process.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+* The live migration is only supported on managed clusters. It is not supported on HyperShift clusters. 
+
+== Information about the OpenShift SDN live migration script
+
+The live migration method for migrating from OpenShift SDN network plugin to the OVN-Kubernetes network plugin is a two-step procedure that contains multiple internal processes that are executed by a wrapper script. When running the two commands, the following events occur:
+
+. The Cluster Network Operator (CNO) sets the `NetworkTypeMigrationInProgress` condition to `True` in the `status` field of the `network.config` custom resource. If there are any unsupported features enabled in the cluster, the CNO sets the `NetworkTypeMigrationInProgress` to `False` and alerts administrators that the migration will not start. Users must disable these unsupported features before starting the live migration.
```
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:26]: Removed this step.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 8 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:24]: Updated to 

The live migration method for migrating from OpenShift SDN network plugin to the OVN-Kubernetes network plugin is a multi-step procedure that resides in the Cluster Network Operator node. It contains multiple internal processes that are executed by a wrapper script. When running the commands, the following events occur:

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 9 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:36]: Changed to: 

1. The Cluster Network Operator checks the current hardware MTU and the cluster MTU to determine a routable MTU during migration. These changes trigger the Machine Config Operator to apply a new machine configuration to each machine config pool and reboot all nodes. After migration, MTU values are reduced by 50 bytes. For example:

     - If the current hardware MTU is `1500`, and the cluster MTU is not specified, the CNO sets the `.spec.migration.mtu` of the `network.operator` CR with the MTU of OVN-Kubernetes: 
         - `.spec.migration.mtu.network.from` is reduced from `1450` to `1400`
         - `.spec.migration.mtu.machine.from` remains `1500`
         
  Which mentions that they're reduced by 50 and provides an example. I can remove the example if suggested.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 11 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:140]: See thread: https://redhat-internal.slack.com/archives/C0590SMCUKV/p1707835117080859

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 10 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:88]: I'm not sure if I'm understanding 100% correctly: 

Do egress IP, egress firewalls, and multicast need to be removed by the administrator before beginning the live procedure? or are they removed automatically?

is the egress router removed automatically? or does it need to be removed by the administrator?

Thanks!

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 12 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:154]: @pliurh Is this command correct? 

oc get network.config <network-config-name> -o=jsonpath='{.status.conditions}'

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 13 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:12]: Added

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 12 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:154]: cc @weliang1

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc]: Will address this when I have clarification on question below.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 12 above)
Comment (weliang1) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:152]: @stevsmit To verify the migration completed successfully through:

`$ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"
                "message": "Network type migration is completed",
$ oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}'
OVNKubernetes`

Comment (pliurh) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:152]: This command can be used `oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .`

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 10 above)
Comment (pliurh) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:86]: The admin shall remove the egress router configuration. The configuration for egress IP, egress firewalls, and multicast can be migrated automatically after the live migration is done. So users don't need to remove them.

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 12 above)
Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc:152]: @weliang1 @pliurh PTAL:

3. After running these commands, the migration process begins. The time it takes to complete the process will vary depending on how many nodes are in your cluster. Enter the following commands to check the status of the `network.config` custom resource:
~~~
$ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"
~~~
~~~
$ oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}'
~~~
~~~
$ oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .
~~~

Comment (weliang1) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc]: @stevsmit Here is the output from oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .

```
$ oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .
[
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationMTUReady"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationTargetCNIAvailable"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationTargetCNIInUse"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationOriginalCNIPurged"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is completed",
    "reason": "NetworkTypeMigrationCompleted",
    "status": "False",
    "type": "NetworkTypeMigrationInProgress"
  }
]
```

You may do not need `oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed" `any more.

Comment (stevsmit) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc]: Removed $ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"

File: modules/nw-ovn-kubernetes-migration-about-live.adoc (hunk 10 above)
Comment (pliurh) [reply modules/nw-ovn-kubernetes-migration-about-live.adoc]: Multicast, egress IP addresses, and egress firewalls are not disabled. They just cannot work as expected.

**Pull request closed without merging**